### PR TITLE
Add BatchNorm layers to CNN in MNIST tutorial for improved training stability

### PR DESCRIPTION
**Pull request by sanepunk**
## What does this PR do?

Integrates `nnx.BatchNorm` into the CNN model used in `docs_nnx/mnist_tutorial.ipynb`. Enables batch-norm-aware behavior with `.train()` and `.eval()` modes to improve convergence and metrics visualization.

### Highlights

- Added `nnx.BatchNorm()` after each convolution  
- Updated training loop to call `model.train()` so running statistics are updated  
- Switched to `model.eval()` in the evaluation loop for deterministic inference  
- Observed smoother loss and accuracy curves in the notebook’s metrics graphs  

### Why?

BatchNorm stabilizes and accelerates training by normalizing activations between layers, leading to better gradient flow and faster convergence.

### Testing

Ran the MNIST tutorial notebook end-to-end and confirmed:

- Training loss decreases more smoothly  
- Validation accuracy improves more quickly  
- Metrics plots clearly reflect these improvements  

---

Closes #<issue-number>

### CODE REVIEW

**Comment by 8bitmp3** [comment]
@IvyZX @jburnim @cgarciae WDYT

**Comment by pccoeit** [comment]
Thanks for the update

**Comment by sanepunk** [comment]
Can you merge this PR? @8bitmp3 @IvyZX @jburnim @cgarciae

**Comment by sanepunk** [comment]
> Can you merge this PR? @8bitmp3 @IvyZX @jburnim @cgarciae 

I’ve made the necessary changes to fix the GitHub Actions failure, everything checks out now.

**Comment by cgarciae** [comment]
@sanepunk looks good! You can fix pre-commit with:

```
pip install pre-commit
pre-commit run --all-files
```

**Comment by sanepunk** [comment]
thank you @cgarciae @8bitmp3 @IvyZX @jburnim for helping me with my first PR to FLAX
